WeatherMonitoringAndReportingService.Tests/DataSourceProcessor/JsonFileProcessorTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿using AutoFixture;
+using Moq;
+using WeatherMonitoringAndReportingService.AppSettings;
+using WeatherMonitoringAndReportingService.Config;
+using WeatherMonitoringAndReportingService.DataSourceProcessor;
+using WeatherMonitoringAndReportingService.DataSourceProcessor.Readers;
+using WeatherMonitoringAndReportingService.DataSourceProcessor.Serializers;
+using WeatherMonitoringAndReportingService.DataSourceProcessor.Writers;
+
+namespace WeatherMonitoringAndReportingService.Tests.DataSourceProcessor;
+
+public class JsonFileProcessorTests
+{
+    private readonly Mock<IFileWriter> _jsonWriterMock;
+    private readonly Mock<IFileSerializer> _jsonSerializerMock;
+    private readonly Mock<IFileReader> _jsonReaderMock;
+    private readonly JSONFileProcessor _jsonProcessor;
+    private readonly Fixture _fixture;
+
+    public JsonFileProcessorTests()
+    {
+        _fixture = new Fixture();
+        _jsonWriterMock = new Mock<IFileWriter>();
+        _jsonSerializerMock = new Mock<IFileSerializer>();
+        _jsonReaderMock = new Mock<IFileReader>();
+        _jsonProcessor = new JSONFileProcessor(_jsonWriterMock.Object, _jsonSerializerMock.Object, _jsonReaderMock.Object);
+
+        _jsonSerializerMock
+            .Setup(serializer => serializer.Serialize(It.IsAny<Dictionary<string, WeatherConfigurationModel>>()))
+            .Returns(_fixture.Create<string>());
+        _jsonReaderMock
+            .Setup(reader => reader.ReadFile(It.IsAny<string>()))
+            .Returns(new Dictionary<string, WeatherConfigurationModel>());
+    }
+
+    [Fact]
+    public void Add_ShouldAddFileAddingNewLine()
+    {
+        // Arrange
+        var name = "RainBot";
+        var weatherConfigurationModel = _fixture.Create<WeatherConfigurationModel>();
+        var path = AppSettingsInitializer.AppSettingsInstance().ConfigFilePath;
+
+        // Act
+        _jsonProcessor.Add(name, new WeatherConfigurationModel(), path);
+
+        // Assert
+        _jsonReaderMock.Verify(readerMock => readerMock.ReadFile(It.IsAny<string>()), Times.Once);
+        _jsonSerializerMock.Verify(
+            serializerMock => serializerMock.Serialize(It.IsAny<Dictionary<string, WeatherConfigurationModel>>()),
+            Times.Once);
+        _jsonWriterMock.Verify(
+            writerMock => writerMock.WriteFile(It.IsAny<string>(), It.IsAny<string>()),
+            Times.Once);
```

Comment:
Whenever you are doing multiple Assertions inside a same test, use a `AssertionScope`

```suggestion
    using (new AssertionScope())
    {
            _jsonReaderMock.Verify(readerMock => readerMock.ReadFile(It.IsAny<string>()), Times.Once);
            _jsonSerializerMock.Verify(
                serializerMock => serializerMock.Serialize(It.IsAny<Dictionary<string, WeatherConfigurationModel>>()),
                Times.Once);
            _jsonWriterMock.Verify(
                writerMock => writerMock.WriteFile(It.IsAny<string>(), It.IsAny<string>()),
                Times.Once);
        }
```